Source/CesiumRuntime/Private/CesiumGltfComponent.cpp

```diff
@@ -91,10 +92,32 @@ using TMeshVector4 = FVector4;
 static uint32_t nextMaterialId = 0;
 
 namespace {
+void destroyHalfLoadedTexture(
+    TUniquePtr<CesiumTextureUtility::LoadedTextureResult>& pHalfLoadedTexture) {
+  if (pHalfLoadedTexture) {
+    CesiumTextureUtility::destroyHalfLoadedTexture(*pHalfLoadedTexture.Get());
+  }
+}
 class HalfConstructedReal : public UCesiumGltfComponent::HalfConstructed {
 public:
-  virtual ~HalfConstructedReal() {}
   LoadModelResult loadModelResult;
+
+  virtual ~HalfConstructedReal() {
+    // TODO: deal with metadata case
```

Comment:
Opened #979 and linked to it from the TODO. No special metadata cleanup needs to go there for now, since encoded metadata doesn't use async texture creation.